app/javascript/components/Greeting.js

```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react"
+import PropTypes from "prop-types"
+import { useDispatch, useSelector } from "react-redux";
+import { fetchMessage } from "../state/message/messageSlice";
+
+
+const Greeting = () => {
+  const dispatch = useDispatch();
+  const { message } = useSelector((state) => state.message);
+  console.log();
+
+  useEffect(() => dispatch(fetchMessage()), [])
+
+  return (
+    <React.Fragment>
+      {message.message_content}
+    </React.Fragment>
+  );
+}
+
+export default Greeting
```

Comment:
- [ ] From the greeting file, You had to add the actions of both `action creator and types` as well as to connect to the rail app to retrieve the random greetings. For example:


 function getGreetings() {
  return dispatch => {
    dispatch({ type: GET_GREETINGS_REQUEST });
    return fetch(`api/grettings.json`)
      .then(response => response.json())
      .then(json => dispatch(getGreetingsSuccess(json)))
      .catch(error => console.log(error));
  };
}; 

export function getGreetingsSuccess(json) {
  return {
    type: GET_GREETINGS_SUCCESS,
    json
  };
};


I see you have implemented the same logic below however try to make it working as expected.